Sources/NIO/DatagramVectorReadManager.swift

```diff
@@ -100,7 +100,7 @@ struct DatagramVectorReadManager {
                 // TODO(cory): almost all of this except for the iovec could be done at allocation time. Maybe we should?
 
                 // First we set up the iovec and save it off.
-                self.ioVector[i] = IOVector(iov_base: bufferPointer.baseAddress! + (i * messageSize), iov_len: messageSize)
+                self.ioVector[i] = IOVector(iov_base: bufferPointer.baseAddress! + (i * messageSize), iov_len: numericCast(messageSize))
```

Comment:
@Lukasa, I tried adding another initializer that takes `iov_len:Int` instead, as you suggested above, but that would then cause [other invocations that used `numericCast()`](https://github.com/apple/swift-nio/blob/main/Sources/NIO/PendingWritesManager.swift#L55) to fail, as they'd get confused over which initializer to use. Since those other `numericCast()` invocations were working fine on Android, changed these three in this pull to use it too.